kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingPredicate.java

```diff
@@ -35,6 +42,24 @@ public class DataSkippingPredicate extends Predicate {
   DataSkippingPredicate(String name, List<Expression> children, Set<Column> referencedCols) {
     super(name, children);
     this.referencedCols = Collections.unmodifiableSet(referencedCols);
+    this.collationIdentifiers = Collections.unmodifiableSet(new HashSet<>());
+  }
+
+  /**
+   * @param name the predicate name
+   * @param children list of expressions that are input to this predicate.
+   * @param collationIdentifier collation identifier used for this predicate
+   * @param referencedCols set of columns referenced by this predicate or any of its child
+   *     expressions
+   */
+  DataSkippingPredicate(
+      String name,
+      List<Expression> children,
+      CollationIdentifier collationIdentifier,
+      Set<Column> referencedCols) {
+    super(name, children, collationIdentifier);
+    this.referencedCols = Collections.unmodifiableSet(referencedCols);
+    this.collationIdentifiers = Collections.singleton(collationIdentifier);
```

Comment:
doesn't this need to look at the children?

Reply:
I see, we only need it for expressions that operate on a data column (==> stats column in the resulting skipping expression). Can you maybe just clarify this in the docs?

Reply:
Also, does this mean we will try to read all the nested fields for a given collation? Should we instead be storing collation + column?